operations/pyroscope/helm/pyroscope/values.yaml

```diff
@@ -186,8 +186,187 @@ agent:
+          targets = concat(discovery.relabel.kubernetes_pods.output)
+
+          rule {
+            source_labels = ["__meta_kubernetes_pod_annotation_profiles_grafana_com_{{.}}_scrape"]
+            action        = "keep"
+            regex         = "true"
+          }
+
+          rule {
+            source_labels = ["__meta_kubernetes_pod_annotation_profiles_grafana_com_{{.}}_port_name"]
+            action        = "drop"
+            regex         = ""
+          }
+
+          rule {
+            source_labels = ["__meta_kubernetes_pod_container_port_name"]
+            target_label  = "__meta_kubernetes_pod_annotation_profiles_grafana_com_{{.}}_port_name"
+            action        = "keepequal"
+          }
+
+          rule {
+            source_labels = ["__meta_kubernetes_pod_annotation_profiles_grafana_com_{{.}}_scheme"]
+            action        = "replace"
+            regex         = "(https?)"
+            target_label  = "__scheme__"
+            replacement   = "$1"
+          }
+
+          rule {
+            source_labels = ["__meta_kubernetes_pod_annotation_profiles_grafana_com_{{.}}_path"]
+            action        = "replace"
+            regex         = "(.+)"
+            target_label  = "__profile_path__"
+            replacement   = "$1"
+          }
+
+          rule {
+            source_labels = ["__address__", "__meta_kubernetes_pod_annotation_profiles_grafana_com_{{.}}_port"]
+            action        = "replace"
+            regex         = "(.+?)(?::\\d+)?;(\\d+)"
+            target_label  = "__address__"
+            replacement   = "$1:$2"
+          }
+        }
+
+        pyroscope.scrape "pyroscope_scrape_{{.}}" {
+          clustering {
+            enabled = true
+          }
+
+          targets    = concat(discovery.relabel.kubernetes_pods_{{.}}_default_name.output, discovery.relabel.kubernetes_pods_{{.}}_custom_name.output)
+          forward_to = [pyroscope.write.pyroscope_write.receiver]
+
+          profiling_config {
+            {{- $currentType := . -}}
+            {{- range $profileTypes }}
+            profile.{{if eq . "cpu"}}process_cpu{{else}}{{.}}{{end}} {
+              enabled = {{if eq . $currentType}}true{{else}}false{{end}}
+            }
+            {{- if ne . (last $profileTypes) }}{{ printf "\n" }}{{ end }}
+            {{- end }}
+          }
+        }
+        {{- end }}
+
+        pyroscope.write "pyroscope_write" {
+          endpoint {
+            {{- if hasKey .Values.pyroscope.components "distributor" }}
+            url = "http://{{ include "pyroscope.fullname" . }}-distributor.{{ .Release.Namespace }}.svc.cluster.local.:{{ (.Values.pyroscope.components.distributor.service).port | default .Values.pyroscope.service.port}}"
+            {{- else }}
+            url = "http://{{ include "pyroscope.fullname" . }}.{{ .Release.Namespace }}.svc.cluster.local.:{{ .Values.pyroscope.service.port }}"
+            {{- end }}
+          }
+        }
```

Comment:
Ah, I just have realized that when the template is rendered in the agent chart, it has no access to `.Values.pyroscope`.

I'm sorry @QuantumEnigmaa, you did well, but we probably can't inject the default config to the agent's config map content (in a straight forward way). I think your former solution (check `create`) is the way to go

Reply:
> when the template is rendered in the agent chart, it has no access to .Values.pyroscope

Yeah you're right, I had totally forgot about that ! I reverted this change back to the first one I did :)